scipy/io/mmio.py

```diff
@@ -281,20 +286,44 @@ def _get_symmetry(a):
         m,n = a.shape
         if m != n:
             return MMFile.SYMMETRY_GENERAL
-        issymm = 1
-        isskew = 1
+        issymm = True
+        isskew = True
         isherm = a.dtype.char in 'FD'
-        for j in range(n):
-            for i in range(j+1,n):
-                aij,aji = a[i][j],a[j][i]
-                if issymm and aij != aji:
-                    issymm = 0
-                if isskew and aij != -aji:
-                    isskew = 0
-                if isherm and aij != conj(aji):
-                    isherm = 0
-                if not (issymm or isskew or isherm):
-                    break
+        
+        if isinstance(a,spmatrix):
+            # check if number of nonzero entrys of lower and upper triangle matrix are equal
+            (row, col) = a.nonzero()
+            if (row < col).sum() != (row > col).sum():
+                return MMFile.SYMMETRY_GENERAL
+            
```

Comment:
A simple counter-example is the following symmetric matrix:
```
cols = np.array([0, 0, 1])
rows = np.array([0, 1, 0])
```
Your code would not recognize the symmetry unlike my code.